File: backend/app/schemas/owner_dashboard.py
from decimal import Decimal
from pydantic import BaseModel


class OwnerDashboardStatsResponse(BaseModel):
    total_properties: int
    approved_properties: int
    pending_properties: int
    rejected_properties: int

    total_bookings: int
    pending_cash_requests: int
    upcoming_bookings: int

    total_revenue: Decimal
    platform_commission: Decimal
    owner_earnings: Decimal
    outstanding_cash_commission: Decimal

    portfolio_rating: Decimal
    total_reviews: int

File: frontend/src/services/ownerDashboard.ts
"use client";
import { apiFetch } from "@/services/api";

export type OwnerDashboardStats = {
  total_properties: number;
  approved_properties: number;
  pending_properties: number;
  rejected_properties: number;
  total_bookings: number;
  pending_cash_requests: number;
  upcoming_bookings: number;
  total_revenue: string | number;
  platform_commission: string | number;
  owner_earnings: string | number;
  outstanding_cash_commission: string | number;
  portfolio_rating: string | number;
  total_reviews: number;
};

export async function getOwnerDashboardStats(): Promise<OwnerDashboardStats> {
  return apiFetch("/owner/dashboard/stats");
}
